src/psyclone/psyir/transformations/inline_trans.py

```diff
@@ -122,47 +134,196 @@ class InlineTrans(Transformation):
         Some of these restrictions will be lifted by #924.
 
     '''
-    def apply(self, node, options=None):
-        '''
+
+    def __init__(self):
+        # List of call-to-subroutine argument indices
+        self._ret_arg_match_list: List[int] = None
+
+        # Call to routine
+        self._call_node: Call = None
+
+        # Routine to be inlined for call
+        self._routine_node: Routine = None
+
+        from psyclone.psyir.tools import CallRoutineMatcher
+
+        self._call_routine_matcher: CallRoutineMatcher = CallRoutineMatcher()
+
+        # If 'True', don't inline if a code block is used within the
+        # Routine.
+        self._option_check_codeblocks: bool = True
+
+        self._option_check_diff_container_clashes: bool = True
+        self._option_check_diff_container_clashes_unres_types: bool = True
+        self._option_check_resolve_imports: bool = True
+        self._option_check_static_interface: bool = True
+        self._option_check_array_type: bool = True
+        self._option_check_unsupported_type: bool = True
+        self._option_check_unresolved_symbols: bool = True
+
+    def set_option(
```

Comment:
1. This list of options is massive - can it be refined/reduced?
2. As mentioned elsewhere, we haven't yet agreed on how to handle options to transformations (#2668) so please stick with the current method for now.